### ISSUE TITLE
Unable to upload complex relevant clause

### ISSUE THREAD

**Issue by alukach** [Cadasta/cadasta-platform#1672]
### Steps to reproduce the error

1. Download example [Customary Rights From](https://docs.google.com/spreadsheets/d/1MHg6iok4SkDxN2NdMVt3P2W9UZe81VxH6CpAz_eUtOY/pub?output=xlsx)
2. Attempt to upload to new project (or project with no data)

### Actual behavior

Won't allow upload:

```
Accepted file types: xls, xlsx

Invalid relevant clause: selected(${what_docs}, 'other')
```

### Expected behavior

Questionniare is accepted

**Comment by dpalomino** [comment]
hi, thanks @alukach for raising this, and thanks @bjohare for the quick analysis!

As it was discussed in slack, we need either to not deploy #1641  in demo/production or allow the complex relevant conditions. They are very widely used in partner's forms.

**Comment by bjohare** [comment]
We could possibly use `lxml's` `etree.XPath` class to parse the `xpath` expression. If it compiles, it should be valid, if not we can throw a `ValidationError`.. needs investigation.

**Comment by dpalomino** [comment]
Thanks @bjohare!

I think you can use [this questionnaire](https://drive.google.com/open?id=1qLQKHsdqBFi_TfW0dpWWkIWarOz4t9RorT--Kf1GHUM) if you want to do some testing. It has a ton of relevant conditions. 

If it is complicated or takes you too much time I would suggest just to revert back the changes. There was not a critical bug. 

Thanks again!

**Comment by oliverroick** [comment]
UPDATE:

We can't validate the conditions using `etree.XPath` because it throws an exception when the condition contains curly brackets:

```
>>> from lxml import etree

>>> etree.XPath("${number_of_lands}>3")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "src/lxml/xpath.pxi", line 423, in lxml.etree.XPath.__init__ (src/lxml/lxml.etree.c:179933)
  
>>> etree.XPath("$number_of_lands>3")
$number_of_lands>3

>>> etree.XPath("$number_of_lands> 3")
$number_of_lands> 3

>>> etree.XPath("$number_of_lands > 3")
$number_of_lands > 3

>>> etree.XPath("selected(${what_docs}, 'other')")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "src/lxml/xpath.pxi", line 423, in lxml.etree.XPath.__init__ (src/lxml/lxml.etree.c:179933)
lxml.etree.XPathSyntaxError: Expected $ for variable reference

>>> etree.XPath("selected($what_docs, 'other')")
selected($what_docs, 'other')
```

On production, we currently have the following patterns:

- `selected(${key}, value)`
- `${key}=|>|>=value`
- `condition and|or condition` (`condition` can be one of the two above)
- `not(condition)` (`condition` can be one of the two above)

I'll look into this a bit further in the coming days. If I can't find a way to solve this, we will revert the breaking commit before the next release.

**Comment by dpalomino** [comment]
Fixed with #1678. Closing the issue.
